File: 0009-palindrome-number/9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x_str = str(x)

        i = 0
        while i < len(x_str) // 2:
            if x_str[i] != x_str[len(x_str) - i-1]:
                return False
            i += 1
        
        return True
# @lc code=end

File: 0009-palindrome-number/9.palindrome-number.rs
/*
 * @lc app=leetcode id=9 lang=rust
 *
 * [9] Palindrome Number
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let xStr = x.to_string();
        let xLen = xStr.chars().count();

        for i in 0..(xLen as f32 / 2.0).floor() as usize {
            if xStr.as_bytes()[i] != xStr.as_bytes()[xLen - i-1] {
                return false;
            } 
        }

        return true;
    }
}
// @lc code=end

